src/client/mod.rs

```diff
@@ -113,8 +150,11 @@ impl<T: tokens::TokenManager> Client<T> {
     /// This URL should be the FQDN plus any namespacing (like `v1`). So if you were running a
     /// bindle server mounted at the v1 endpoint, your URL would look something like
     /// `http://my.bindle.com/v1/`. Will return an error if the URL is not valid
-    pub fn new(base_url: &str, token_manager: T) -> Result<Self> {
-        ClientBuilder::default().build(base_url, token_manager)
+    ///
+    /// This requires `KeyRing` to be wrapped in an `Arc` for efficiency sake. `KeyRing`s can be
+    /// quite large and so we require it to be wrapped in an Arc to avoid a large clone cost
```

Comment:
I was gonna ask why you used an Arc. This makes total sense.

Reply:
Yeah I figured it would be best to expose this directly in the API as an `Arc` because there are other places you could be needing the same keyring in code. This accounts for that so you can clone it willy nilly if you so desire